app/src/main/java/com/conupods/OutdoorMaps/View/Directions/ModeSelectActivity.java

```diff
@@ -0,0 +1,293 @@
+        toButton.setText("To: " + toLongName);
+
+        Button walkingBTN = (Button) findViewById(R.id.modeSelect_walkingButton);
+        walkingBTN.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                launchModeSelectIntent(TravelMode.WALKING);
+            }
+        });
+
+        Button drivingBTN = (Button) findViewById(R.id.modeSelect_drivingButton);
+        drivingBTN.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                launchModeSelectIntent(TravelMode.DRIVING);
+            }
+        });
+
+        Button transitBTN = (Button) findViewById(R.id.modeSelect_transitButton);
+        transitBTN.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                launchModeSelectIntent(TravelMode.TRANSIT);
+            }
+        });
+    }
+
+    public void onClickSetOrigin(View view) {
+        Intent modeSelectIntent = new Intent(this, FinalizeSearchActivity.class);
+        loadLocationsIntoIntent(modeSelectIntent);
+        startActivity(modeSelectIntent);
+    }
+
+    public void onClickSetDestination(View view) {
+        Intent modeSelectIntent = new Intent(this, SearchActivity.class);
+        loadLocationsIntoIntent(modeSelectIntent);
+        startActivity(modeSelectIntent);
+    }
+
+    private void unpackIntent(Intent intent) {
+        fromLongName = intent.getStringExtra("fromLongName");
+
+        if (fromLongName.equals("Current Location")) {
+            LocationManager lm = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+
+            if (ActivityCompat.checkSelfPermission(this,
+                    Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED &&
+                    ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+                // TODO: Consider calling
+                //    ActivityCompat#requestPermissions
+                // here to request the missing permissions, and then overriding
+                //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+                //                                          int[] grantResults)
+                // to handle the case where the user grants the permission. See the documentation
+                // for ActivityCompat#requestPermissions for more details.
+                return;
+            }
+            Location location = lm.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+            if(location == null) {
+                location = lm.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+                if(location == null) {
+                    mOrigin = new LatLng(45.4973, -73.5790);    // Hall building
```

Comment:
Dont we have that info stored somewhere, to make everything unified, u should use that

Reply:
this information is stored within the first activitys own map. On a new map, and so a lighter version with the same global permissions is used